8546 connect order to motion revamped

JayFlexy opened the pull request:
See #8546

zacharybrown204 committed:
8546 Adding order columns to docket entry

JayFlexy committed:
8546: wip add comments for points of interests

JayFlexy committed:
8546: wip more points of interest

JayFlexy committed:
8546: wip add checkbox

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion

zacharybrown204 committed:
8546 - WIP API updates to add dispositions to motions

JayFlexy committed:
8546: connect checkbox

zacharybrown204 committed:
8546 WIP - API side updates

Added DB updates to file and serve interactor
Added updates to getCasesByDocketNumbers.

zacharybrown204 committed:
8546 - Adding typescript types to document metadata for the file interactors

JayFlexy committed:
8546: add new motion disposition option

JayFlexy committed:
8546: wip fix data connections

JayFlexy committed:
8546: wip debug order modal failing to confirm

zacharybrown204 committed:
8546 - Motions + Orders Connecting UI to API

JayFlexy committed:
8546: add Dispotion to motion, filter previous disposed motions

zacharybrown204 committed:
8546 - Now able to add multiple motions to the order

JayFlexy committed:
8546: branch reconstruction wip

zacharybrown204 committed:
8546

Tweaks to adding multiple motions to order

JayFlexy committed:
8546: robustify index grabbing function to prevent faulty indices

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

codyseibert commented on shared/src/business/utilities/getFormattedCaseDetail.ts:
```diff
@@ -50,7 +50,7 @@ export const computeIsNotServedDocument = ({ formattedEntry }) => {
   );
 };
-
+// TODO (#8546): update disposed motions here
```
left over comments?

codyseibert commented on web-api/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.ts:
```diff
@@ -69,6 +73,11 @@ export const fileAndServeCourtIssuedDocument = async (
   });
 
+  console.log(`!!!!!!!!!!!!!!!!!!!!!!`);
+  console.log('subjectCase:', subjectCase);
```
left over logs

zacharybrown204 committed:
8546 - small style update

JayFlexy committed:
8546: All caps dispositions

zacharybrown204 committed:
Merge branch '8546-connect-order-to-motion-revamped' of https://github.com/ustaxcourt/ef-cms into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546

Updating file & file and serve to account for cosolidated cases.

JayFlexy committed:
8546: add ability to hold docket entry relationships

JayFlexy committed:
8546: some comment and log cleanup

zacharybrown204 committed:
8546 Docket entry relations

Changing docket entry relation structure to be more generic.
Allowing for order documents to point back to motions as well.

zacharybrown204 committed:
8546 Adding line breaks when listing related docket entries

zacharybrown204 committed:
8546 - Updating table and column names

zacharybrown204 committed:
8546 - Added form logic when adding motions to order

Motions already selected will now be excluded from other dropdowns when associating with orders

JayFlexy committed:
rm old log statement

JayFlexy committed:
8546: fix link to child cases

JayFlexy committed:
8546: wip

zacharybrown204 committed:
8546 - Refacotring related docket entries table

Consolidating docket number columns into single column

zacharybrown204 committed:
Merge branch '8546-connect-order-to-motion-revamped' of https://github.com/ustaxcourt/ef-cms into 8546-connect-order-to-motion-revamped

JayFlexy committed:
8546: wip, start validation work on new branch

zacharybrown204 committed:
8546 - Updating edit court issued document view to handle motions + orders

JayFlexy committed:
Merge branch '8546-connect-order-to-motion-revamped' into 8546-connect-order-to-motion-revamped-validation-work

JayFlexy committed:
8546: add validation for court issued docket entry (order)

JayFlexy committed:
8546: rm outdated comment

zacharybrown204 committed:
8546 - Adding validation for motions & orders form

zacharybrown204 committed:
8546 - Add validation to edit form

zacharybrown204 committed:
8546 - Updating types in file interactor

zacharybrown204 committed:
8546 Typescript updates

JayFlexy committed:
8546: wip cypress test

JayFlexy committed:
8546: wip cypress tests

JayFlexy committed:
8546: rm CourtIssueDocumentBase

zacharybrown204 committed:
8546 - Typescript updates - fixed a circular dependency

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - unit test updates & typescript fixes

zacharybrown204 committed:
8546 - Adding happy path tests for relating documents

JayFlexy committed:
8546: wip cypress debugging

JayFlexy committed:
8546: cypress test - order affecting a docket entry

zacharybrown204 committed:
8546 - Re-adding court issued document base as partial document

zacharybrown204 committed:
8546 - Fixing failing tests in github actions

zacharybrown204 committed:
8546 - Fixes for validation updates in shared tests

zacharybrown204 committed:
8546 - Fixing integration and cypress tests

zacharybrown204 committed:
8546 - Fixing typescript errors and unit tests.

zacharybrown204 committed:
8546 - Fixing cypress test and adding more test coverage

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

codyseibert commented on shared/src/business/entities/DocketEntry.ts:
```diff
@@ -186,6 +187,10 @@ export class DocketEntry extends JoiValidationEntity {
   public strickenByUserId?: string;
+  // public orderDocketEntryId?: string;
+  // public motionDisposition?: string; // 'GRANTED' | 'DENTIED' | 'PARTIAL';
+  public affectedDocketEntries?: DocketEntryRelation[];
```
I'm not sure I understand the term "affected".  what is the intent of these new fields?  affected by what?

JayFlexy replied:
The new field `affectedDocketEntries` refer to docket entries that are being affected by a different docket entry. So in this case we create an order that affects a motion. Orders and motions are but one type of inter-relation amongst docket entries but we were attempting to make the code generic so that future relations between docket entries are not restricted to order/motion verbiage.

codyseibert commented on web-client/src/views/DocketRecord/FilingsAndProceedings.tsx:
```diff
@@ -143,6 +156,39 @@ export const FilingsAndProceedings = connect<
+            ).map(entry => {
+              return (
+                <>
+                  <br></br>
```
I'm not sure the indent of the <br> but I'd recommend using gap or margin or padding if the goal is to space things out instead of just br tags

JayFlexy replied:
Will review this to see if there are better options than `<br>` tag. I do vaguely remember this being a finicky part of the UI where margin/padding tinkering never really gave what we were intending.

codyseibert commented on web-api/src/persistence/postgres/cases/getCasesByDocketNumbers.ts:
```diff
@@ -309,9 +378,77 @@ export async function getDocketEntriesOnCases(
-      .selectFrom('dwDocketEntry')
+      .with('affectedDocketEntries', db =>
+        db
+          .selectFrom('dwDocketEntryRelatedDocketEntry')
```
Just curious, but has an EXPLAIN statement been executd on this query to check it's performance and verify no indexes are needed.  The where('served', 'is', true) seems to be used in this CTE but won't that loop over a ton of docket entries? I probably don't understand the intent.  I would think the CTEs should at least do an IN clause by docket number so that it only has to fetch the related cases instead of needing to grab all the rows in the docket entry table, which is I think 10 million last time I remembered.

.with('affectedDocketEntries', db =>
        db
          .selectFrom('dwDocketEntryRelatedDocketEntry')
          .where('served', 'is', true)
          .where('docketNumber', 'in', docketNumbers) // ✅ Add filter here!
          
          
additionally, the 
join
          .onRef(
            eb => eb.cast('affectedByDocketEntries.docketEntryId', 'varchar'),
claude is hinting that this cast might prevent the db engine from using the index correctly which might also cause some performance issues.

zacharybrown204 committed:
8546 - Few DB updates based off PR feedback

JayFlexy committed:
8546: rm unused check

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

JayFlexy committed:
8546: comment cleanup

zacharybrown204 committed:
8546 PR updates

Removing misc comments
Using explicit imports from lodash
Extracting common code to helper function
Reworking cypress test to be more idempotent

zacharybrown204 committed:
8546 - Add test to cover updating docket entry with affected docket entries.

JayFlexy committed:
8546: add unit test coverage for case motions

JayFlexy committed:
Merge remote-tracking branch 'origin/staging' into 8546-connect-order-to-motion-revamped

JayFlexy committed:
8546: rm comments

JayFlexy committed:
8546: address feedback, cleanup

JayFlexy committed:
Merge remote-tracking branch 'origin/staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - Adding additional unit tests

JayFlexy committed:
Merge remote-tracking branch 'origin/staging' into 8546-connect-order-to-motion-revamped

JayFlexy committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 Spacing adjustments

JayFlexy committed:
Merge remote-tracking branch 'origin/staging' into 8546-connect-order-to-motion-revamped

JayFlexy committed:
Merge remote-tracking branch 'origin/staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - Adding related docket entries to public views.

zacharybrown204 committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546

Refactoring array logic into helper

zacharybrown204 committed:
8546

Missed a null check in previous commit.

zacharybrown204 committed:
8546 - cleaning up branch

zacharybrown204 committed:
8546 - Removing console log

zacharybrown204 committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
Merge branch '8546-connect-order-to-motion-revamped' of https://github.com/ustaxcourt/ef-cms into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - Fixing validation issue & refactoring out unnecessary variable

zacharybrown204 committed:
Merge remote-tracking branch 'origin/staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
tsconfig update - fix failing type error

zacharybrown204 committed:
8546 - Updates addressing court feedback

zacharybrown204 committed:
Renaming single letter variable names

zacharybrown204 committed:
8546 - typescript & test fixes

zacharybrown204 committed:
8546 - Fixing fialing cypress test

zacharybrown204 committed:
8546 - text update

zacharybrown204 committed:
8546 - Adding a null check to allow dropdowns to be clearable.

jimlerza committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

zacharybrown204 committed:
8546 - Adding docket relationships to printed docket record

Fixed a bug where sealing a docket entry will not load relationship information.
Fixed some errors being thrown on the docket record screen. One where there was an empty space in the table that react didn't like. Another where we had a map without keys.

zacharybrown204 committed:
8546 - Consolidating docket entry queries

zacharybrown204 committed:
8546 - fixing issue with docketEntryBaseQuery

Refactored related docket entry code in generateDocketRecordPdfInteractor into new function

zacharybrown204 committed:
8546 - Changing relationship verbiage on public page & fixing key errors

zacharybrown204 committed:
8546 - Fixing issue with terminal user and public sorting

zacharybrown204 committed:
8546 - Fixing link in public document links.

zacharybrown204 committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

jimlerza committed:
Merge branch 'staging' into 8546-connect-order-to-motion-revamped

jimlerza commented on web-api/src/business/useCases/docketEntry/updateDocketEntryMetaInteractor.ts:
```diff
@@ -209,3 +222,36 @@ export const updateDocketEntryMetaInteractor = withLocking(
+    if (added.length > 0 || updated.length > 0) {
+      await upsertDocketEntryRelatedEntries({
+        orderDocketEntry: docketEntryMeta,
+        motionDocketEntries: _.concat(added ?? [], updated ?? []).map(m => ({
```
Can you import just `concat` instead of importing all of lodash in this interactor?

jimlerza commented on web-api/src/persistence/postgres/docketEntries/commonQueries.ts:
This is a great idea! ❤️

jimlerza commented on web-client/src/views/CourtIssuedDocketEntry/CourtIssuedDocketEntry.tsx:
```diff
@@ -18,6 +18,9 @@ import { reactSelectValue } from '@web-client/ustc-ui/Utils/documentTypeSelectHe
 import { sequences } from '@web-client/presenter/app.cerebral';
 import { state } from '@web-client/presenter/app.cerebral';
 import React from 'react';
+import _ from 'lodash';
```
Here's another instance where we can probably import just the methods we need.

jimlerza commented on web-client/src/views/DocketRecord/FilingsAndProceedings.tsx:
```diff
@@ -6,12 +6,20 @@ import { sequences } from '@web-client/presenter/app.cerebral';
 import { state } from '@web-client/presenter/app.cerebral';
 import React from 'react';
 import classNames from 'classnames';
+import _ from 'lodash';
```
Editor tells me this one is unused.

jimlerza commented on web-client/src/views/ModalDialog.tsx:
```diff
@@ -81,6 +81,7 @@ export const ModalDialog = ({
   };
 
   const runConfirmSequence = evt => {
+    console.log(`runConfirmSequence was triggered`);
```
Do we want to keep this or was this for debugging?

jimlerza reviewed (changes requested):
Looks great! I left a couple of nits related to how lodash is being imported.

zacharybrown204 committed:
8546 - Lodash updates
